presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveCaching.java

```diff
@@ -95,6 +98,18 @@ private void testReadFromTable(String tableNameSuffix)
         query("DROP TABLE " + nonCachedTableName);
     }
 
+    private void initializeRubixStats()
+    {
+        // TODO this is needed for now because Rubix JMX statistis are only registered lazily
```

Comment:
I think we would get empty QueryResult for stats when FS is not yet initialised. We could handle that case and return `0` from `getRemoteReads` and other methods to avoid running queries to warmup.

Reply:
Actually the queries fail as the table does not have expected columns yet. We can work around that in test, reacting for query failure. But I think proper fix would be on Rubix side. So all the jmx stats are registered as soon as Rubix is started up.

Reply:
Lazy initialisation of stats in Rubix came out of a requirement. We now return stats from Rubix daemons too and needed `Configuration` object to talk to those daemons. 
For embedded mode, one way could be Presto passes the Configuration object in `setLocalBookKeeper` call (change needed on Rubix side) and Rubix initialises stats using this.

Reply:
PR idea https://github.com/qubole/rubix/